src/pages/host/getHostEventString.tsx

```diff
@@ -0,0 +1,285 @@
+import React from "react";
+import { HostEventLogData } from "gql/generated/types";
+import styled from "@emotion/styled/macro";
+import { stringifyNanoseconds } from "utils/string";
+import Code from "@leafygreen-ui/code";
+import { StyledLink } from "components/styles";
+import { getUiUrl } from "utils/getEnvironmentVariables";
+import { Collapse } from "antd";
+const { Panel } = Collapse;
+
+const getTerminationString = (monitorOp: string) => {
+  switch (monitorOp) {
+    case `decommissioned`:
+      return `host was decommissioned.`;
+    case `idle`:
+      return `host was idle.`;
+    case `excess`:
+      return `host was decommissioned.`;
+    case `provision_timeout`:
+      return `host took too long for provisioning to complete.`;
+    case `provision_failed`:
+      return `provisioning failed.`;
+    case `expired`:
+      return `expiration time passed.`;
+    default:
+      return `${monitorOp}`;
+  }
+};
+
+export const getHostEventString = (
+  eventType: string,
+  data: HostEventLogData
+) => {
+  switch (eventType) {
+    case `HOST_CREATED`:
+      return `Host created`;
+    case `HOST_AGENT_DEPLOY_FAILED`:
+      return `New agent deploy failed`;
+    case `HOST_PROVISION_ERROR`:
+      return `Host encountered error during provisioning`;
+    case `HOST_STARTED`:
+      return `Host start attempt ${data.successful ? "succeeded" : ""}`;
+    case `HOST_STOPPED`:
+      return `Host stop attempt ${data.successful ? "succeeded" : ""}`;
+    case `HOST_MODIFIED`:
+      return `Host modify attempt ${data.successful ? "succeeded" : ""}`;
+    case `HOST_AGENT_DEPLOYED`:
+      return (
+        <div>
+          {" "}
+          Agent deployed {data.agentRevision ? "with revision" : ""}{" "}
+          <b>{data.agentRevision}</b> {data.agentBuild ? " from " : ""}
+          <b>{data.agentBuild}</b>{" "}
+        </div>
+      );
+    case `HOST_AGENT_MONITOR_DEPLOYED`:
+      return (
+        <div>
+          {" "}
+          Agent monitor deployed {data.agentRevision
+            ? "with revision"
+            : ""}{" "}
+          <b>{data.agentRevision}</b>{" "}
+        </div>
+      );
+    case `HOST_AGENT_MONITOR_DEPLOY_FAILED`:
+      return `New agent monitor deploy failed`;
+    case `HOST_JASPER_RESTARTING`:
+      return (
+        <div>
+          {" "}
+          Jasper service marked as restarting {data.user ? "by" : ""}{" "}
+          <b>{data.user}</b>{" "}
+        </div>
+      );
+    case `HOST_JASPER_RESTARTED`:
+      return (
+        <div>
+          {" "}
+          Jasper service restarted with revision<b>
+            {data.jasperRevision}
+          </b>{" "}
+        </div>
+      );
+    case `HOST_JASPER_RESTART_ERROR`:
+      return (
+        <div>
+          {" "}
+          Host encountered error when restarting Jasper service
+          {data.logs ? (
+            <HostEventLog title="Provisioning logs" logs={data.logs} isCode />
```

Comment:
This thought is still a bit half-baked in my mind so feel free to disregard if you've already tried this. To avoid having to conditionally return the same HostEventLog element in a bunch of these cases (and maybe even the div), what if the switch/case logic is its own function that returns an object with the text and additional logs, and a different function actually returns the jsx to display the event?

Reply:
discussed over slack